eng/tools/spec-gen-sdk-runner/src/change-files.ts

```diff
@@ -0,0 +1,96 @@
+import path from "node:path";
+import {
+  execAsync,
+  getChangedFiles,
+  searchRelatedParentFolders,
+  searchRelatedTypeSpecProjectBySharedLibrary,
+  searchSharedLibrary,
+} from "./utils.js";
+import { logMessage } from "./log.js";
+import { SpecGenSdkCmdInput } from "./types.js";
+
+const readmeMdRegex = /^readme.md$/;
+const typespecProjectRegex = /^tspconfig.yaml$/;
+const typespecProjectSharedLibraryRegex = /[^/]+\.Shared/;
+
+type ChangedSpecs = {
+  [K in "readmeMd" | "typespecProject"]?: string;
+} & {
+  specs: string[];
+};
+
+export async function detectChangedSpecConfigFiles(
+  commandInput: SpecGenSdkCmdInput,
+): Promise<ChangedSpecs[]> {
+  const prChangedFiles: string[] = getChangedFiles(commandInput.localSpecRepoPath) ?? [];
+  if (prChangedFiles.length === 0) {
+    logMessage("No files changed in the PR");
+  }
+  logMessage(`Changed files in the PR: ${prChangedFiles.length}`);
+  const fileList = prChangedFiles.filter((p) => !p.includes("/scenarios/"));
```

Comment:
What is special about "scenarios" here?

Reply:
We intend to exclude SDK generation for changes coming from the `scenarios` folder.

Reply:
There are also other paths like examples should those be included? I'm mostly asking this question as it seems a little odd to only figure out this one folder type if what you are looking for is json files.

Reply:
Changes in the `examples` folder would need to trigger the automation run because samples would be used in the SDK test/sample generation step. I am not aware of any other folders we might have in the specification folder. I will also add 'json' to the filter.